### ISSUE TITLE
Parameter properties trigger `no-useless-constructor`

### ISSUE THREAD

**Issue by o-nix** [eslint/typescript-eslint-parser#546]
**What version of TypeScript are you using?**

3.1.6

**What version of `typescript-eslint-parser` are you using?**

5.8.0 + `standardjs` config.

**What code were you trying to parse?**

````
constructor (private readonly appService: AppService) { }
````

**What did you expect to happen?**

Should be valid code.

**What happened?**

It currently triggers `no-useless-constructor`.

https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties

> Parameter properties are declared by prefixing a constructor parameter with an accessibility modifier or readonly, or both. Using private for a parameter property declares and initializes a private member; likewise, the same is done for public, protected, and readonly.

So this shouldn't be invalid.

**Comment by mysticatea** [comment]
Thank you for this report.

That syntax doesn't exist in JavaScript, so the `no-useless-constructor` rule, one of ESLint core rule that verifies JavaScript, cannot realize the syntax.
Please open issue on `eslint-plugin-typescript` to request a rule to check it for TypeScript.

....Hm, https://github.com/nzakas/eslint-plugin-typescript/issues/101 looks like the one.

**Comment by o-nix** [comment]
Oh, thank you for clarification!
